File: testProject/src/main/java/com/example/testProject/Services/BookService.java
package com.example.testProject.Services;

import java.util.List;
import java.util.Optional;

import com.example.testProject.DTO.Book;

public interface BookService {
	public List<Book> getAllBooks();

	public Optional<Book> getBookById(Long Id);

	public Book createNewBook(Book newBook);

	public Book updateBook(Book updatedBook, Long Id);

	public void deleteBook(Long id);
}

File: test-app/src/services/BookService.js
import axios from 'axios';
import { apiUrl } from '../constants';

class BookService {
    constructor() {
        if (localStorage.getItem('jwtToken') === null) { this.token = '' }
        else {
            this.token = localStorage.getItem('jwtToken');
        }

    }

    getAllBooks() {
        return axios.get(`${apiUrl}/book`, { headers: { authorization: `Bearer ${this.token}` } });
    }

    createNewBook(newBook) {
        return axios.post(`${apiUrl}/book`, newBook, { headers: { authorization: `Bearer ${this.token}` } });
    }

    deleteBook(id) {
        return axios.delete(`${apiUrl}/book/${id}`, { headers: { authorization: `Bearer ${this.token}` } });
    }

    updateBook(id, updatedBook) {
        return axios.put(`${apiUrl}/book/${id}`, updatedBook, { headers: { authorization: `Bearer ${this.token}` } });
    }

    getBookById(id) {
        return axios.get(`${apiUrl}/book/${id}`, { headers: { authorization: `Bearer ${this.token}` } });
    }
}


export default new BookService();